pyqtgraph/functions.py

```diff
@@ -253,45 +253,26 @@ def mkColor(*args):
             c = args[0]
             if len(c) == 1:
                 try:
-                    return Colors[c]
+                    return QtGui.QColor(Colors[c])  # return copy
                 except KeyError:
                     raise ValueError('No color named "%s"' % c)
-            have_alpha = len(c) in [5, 9] and c[0] == '#'  # "#RGBA" and "#RRGGBBAA"
-            if not have_alpha:
-                # try parsing SVG named colors, including "#RGB" and "#RRGGBB".
-                # note that QColor.setNamedColor() treats a 9-char hex string as "#AARRGGBB".
```

Comment:
We could convert pyqtgraph's internal `#RRGGBBAA` hex string format to Qt's `#AARRGGBB` format by simply reshuffling the string, then the code could go through the QColor(string) constructor in *all* cases where the input is a string.

If length is '#' + 4, then the format is RGBA and needs to be expanded manually by e.g. the neat `"".join([x + x for x in c])` construction used below.

After that, or when length is '#' + 8, then reshuffle `c = '#' + c[8:10] + c[1:7]`

Then, `QColor(c)` should handle everything.